[Bug]: Failed to start MO when set logService.replicas = 1 in MatrixOneCluster CRD

### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

- Version or commit-id (e.g. v0.1.0 or 8b23a93):sha-c939c87
- Hardware parameters:
- OS type:
- Others:


### Actual Behavior

Failed to start log pod when set logService.replicas = 1 in MatrixOneCluster CRD

### Expected Behavior

MO success start all pods.

### Steps to Reproduce

mo.yaml:
```
apiVersion: core.matrixorigin.io/v1alpha1
kind: MatrixOneCluster
...
  logService:
    replicas: 1
...
```
kubectl apply -f mo.yaml

### Additional information

_No response_

when I set `logService.replicas=1`,  I found following error. after ask @volgariver6 for help, he told me that we should set `gossip-allow-self-as-seed = true` when these is only one replicas, I have test it and it works, I will promote a pr to fix it and add e2e test.

```/mo-service -cfg /tmp/tmp.5TrGUVNHI2
{"level":"ERROR","time":"2023/03/22 03:13:50.649584 +0000","name":"log-service",
"caller":"logservice/service.go:108",
"msg":"failed to create log store","uuid":"00000000-0000-0000-0000-000000000000",
"error":"no valid seed node",
"errorVerbose":"no valid seed node"

 attached stack trace
 -- stack trace:
|github.com/lni/dragonboat/v4/config.(*GossipConfig).Validate
\t/go/pkg/mod/github.com/matrixorigin/dragonboat/v4@v4.0.0-20221226075848-d266f8c2420c/config/config.go:1011
github.com/lni/dragonboat/v4/config.(*NodeHostConfig).Validate\n  | \t/go/pkg/mod/github.com/matrixorigin/dragonboat/v4@v4.0.0-20221226075848-d266f8c2420c/config/config.go:575\n  | github.com/lni/dragonboat/v4.NewNodeHost\n  | \t/go/pkg/mod/github.com/matrixorigin/dragonboat/v4@v4.0.0-20221226075848-d266f8c2420c/nodehost.go:308\n  | github.com/matrixorigin/matrixone/pkg/logservice.newLogStore\n  | \t/go/src/github.com/matrixorigin/matrixone/pkg/logservice/store.go:146\n  | github.com/matrixorigin/matrixone/pkg/logservice.NewService\n  | \t/go/src/github.com/matrixorigin/matrixone/pkg/logservice/service.go:106\n  | main.startLogService\n  | \t/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:231\n  | main.startService\n  | \t/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:148\n  | main.main\n  | \t/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:95\n  | runtime.main\n  | \t/usr/local/go/src/runtime/proc.go:250\n  | runtime.goexit\n  | \t/usr/local/go/src/runtime/asm_amd64.s:1594\nWraps: (2) no valid seed node\nError types: (1) *withstack.withStack (2) *errutil.leafError","stacktrace":"github.com/matrixorigin/matrixone/pkg/logservice.NewService\n\t/go/src/github.com/matrixorigin/matrixone/pkg/logservice/service.go:108\nmain.startLogService\n\t/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:231\nmain.startService\n\t/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:148\nmain.main\n\t/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:95\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
panic: no valid seed node

goroutine 1 [running]:
main.startLogService(0xc003170d00, 0xc000188000?, {0x2cbb560, 0xc00363fe78})
	/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:234 +0x249
main.startService(0x7ffd026a9b13?, 0x4?)
	/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:148 +0xfa
main.main()
	/go/src/github.com/matrixorigin/matrixone/cmd/mo-service/main.go:95 +0x298```

--- a/pkg/controllers/logset/configmap.go
+++ b/pkg/controllers/logset/configmap.go
@@ -138,6 +138,10 @@ func buildConfigMap(ls *v1alpha1.LogSet) (*corev1.ConfigMap, error) {
 	conf.Set([]string{"logservice", "deployment-id"}, deploymentID(ls))
 	conf.Set([]string{"logservice", "logservice-listen-address"}, fmt.Sprintf("0.0.0.0:%d", logServicePort))
 	conf.Set([]string{"hakeeper-client", "discovery-address"}, fmt.Sprintf("%s:%d", discoverySvcAddress(ls), logServicePort))
+	if ls.Spec.Replicas == 1 {
+		// logservice cannot start up if this gossip option is not set when there is only one replica
+		conf.Set([]string{"logservice", "gossip-allow-self-as-seed"}, true)
+	}
 	s, err := conf.ToString()
 	if err != nil {
 		return nil, err

--- a/test/e2e/logset_test.go
+++ b/test/e2e/logset_test.go
@@ -154,4 +154,73 @@ var _ = Describe("MatrixOneCluster test", func() {
 			return nil
 		}, teardownClusterTimeout, pollInterval).Should(Succeed())
 	})
+
+	It("Should start logset service successfully when logset replicas is 1", func() {
+		By("Create logset")
+		l := &v1alpha1.LogSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: env.Namespace,
+				Name:      "log-" + rand.String(6),
+			},
+			Spec: v1alpha1.LogSetSpec{
+				LogSetBasic: v1alpha1.LogSetBasic{
+					PodSet: v1alpha1.PodSet{
+						Replicas: 1,
+						MainContainer: v1alpha1.MainContainer{
+							Image: fmt.Sprintf("%s:%s", moImageRepo, moVersion),
+						},
+					},
+					Volume: v1alpha1.Volume{
+						Size: resource.MustParse("100Mi"),
+					},
+					SharedStorage: v1alpha1.SharedStorageProvider{
+						FileSystem: &v1alpha1.FileSystemProvider{
+							Path: "/test",
+						},
+					},
+					StoreFailureTimeout: &metav1.Duration{Duration: 2 * time.Minute},
+				},
+			},
+		}
+		Expect(kubeCli.Create(ctx, l)).To(Succeed())
+
+		Eventually(func() error {
+			if err := kubeCli.Get(ctx, client.ObjectKeyFromObject(l), l); err != nil {
+				logger.Errorw("error get logset status", "logset", l.Name, "error", err)
+				return err
+			}
+			if !recon.IsReady(&l.Status.ConditionalStatus) {
+				logger.Infow("wait logset ready", "logset", l.Name)
+				return errWait
+			}
+			return nil
+		}, createLogSetTimeout, pollInterval).Should(Succeed())
+
+		By("Teardown logset")
+		Expect(kubeCli.Delete(ctx, l)).To(Succeed())
+		Eventually(func() error {
+			err := kubeCli.Get(ctx, client.ObjectKeyFromObject(l), l)
+			if err == nil {
+				logger.Infow("wait logset teardown", "logset", l.Name)
+				return errWait
+			}
+			if !apierrors.IsNotFound(err) {
+				logger.Errorw("unexpected error when get logset", "logset", l, "error", err)
+				return err
+			}
+			podList := &corev1.PodList{}
+			err = kubeCli.List(ctx, podList, client.InNamespace(l.Namespace))
+			if err != nil {
+				logger.Errorw("error list pods", "error", err)
+				return err
+			}
+			for _, pod := range podList.Items {
+				if strings.HasPrefix(pod.Name, l.Name) {
+					logger.Infow("Pod that belongs to the logset is not cleaned", "pod", pod.Name)
+					return errWait
+				}
+			}
+			return nil
+		}, teardownClusterTimeout, pollInterval).Should(Succeed())
+	})
 })

